server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1921,7 +1921,7 @@ public void close(String reason, boolean flushEngine, boolean deleted) throws IO
 
                     if (deleted && engine != null && isPrimaryMode() && isRemoteTranslogEnabled()) {
                         // Translog Clean up
-                        assert engine instanceof InternalEngine;
+                        assert ((engine instanceof InternalEngine) || state == IndexShardState.CLOSED);
```

Comment:
this code is different on main 

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L1917

Need to check why this assertion was added. For closed indices, the engine would be NoOpEngine